tokio-fs/src/file/mod.rs

```diff
@@ -182,6 +422,29 @@ impl File {
     /// This function will return an error if the user lacks permission change
     /// attributes on the underlying file. It may also return an error in other
     /// os-specific unspecified cases.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # extern crate tokio;
+    /// use tokio::prelude::*;
+    /// 
+    /// fn main() {
+    ///     let task = tokio::fs::File::create("foo.txt")
+    ///         .and_then(|file| file.metadata())
+    ///         .map(|file_metadata| {
+    ///             let (mut file, metadata) = file_metadata;
```

Comment:
I might collapse this line and pull out both `file` and `metadata` right from the closure params.